File: output/CodeGen/avatar/Java/Python/codeforces_302_A.py
import java.io.IOException ; import java.util.Scanner ; def main ( ):
    sc = Scanner ( System.in )
    n = sc.nextInt ( )
    k = sc.nextInt ( )
    arr = [ 0 ] * n
    for i in range ( 0, n ):
        arr [ i ] = sc.nextInt ( )
    res = [ ]
    o = 0
    e = 0
    for i in range ( 0, n ):
        if arr [ i ] == 1:
            o += 1
        else:
            e += 1
    for i in range ( 0, k ):
        l = sc.nextInt ( )
        r = sc.nextInt ( )
        if ( ( r - l + 1 ) % 2 == 1 ):
            res.append ( "0\n" )
        else:
            if ( ( r - l + 1 ) / 2 <= o and ( r - l + 1 ) / 2 <= e ):
                res.append ( "1\n" )
            else:
                res.append ( "0\n" )
    print ( "".join ( res ) )
    sc.close ( )

File: output/CodeLlama/avatar/Python/Java/codeforces_302_A.java
import java.util.Scanner;
public class codeforces_302_A {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int m = sc.nextInt();
        int sa = 0;
        for (int i = 0; i < n; i++) {
            if (sc.next().equals("-")) {
                sa++;
            }
        }
        sa = Math.min(n - sa, sa);
        for (int i = 0; i < m; i++) {
            int a = sc.nextInt();
            int b = sc.nextInt();
            b -= a;
            if (b % 2 == 1 && b <= sa * 2) {
                System.out.println("1");
            } else {
                System.out.println("0");
            }
        }
    }
}